File: planners/java/JPlanner/src/main/java/JPlanner/Action.java
package JPlanner;

import java.util.Set;

/**
 *
 * @author Andrea Micheli
 */
public class Action {
    private String name;
    private Set<String> precondition;
    private Set<String> positiveEffect;
    private Set<String> negativeEffect;

    public Action(String name, Set<String> precondition, Set<String> positive_effect, Set<String> negative_effect) {
        this.name = name;
        this.precondition = precondition;
        this.positiveEffect = positive_effect;
        this.negativeEffect = negative_effect;
    }

    public String getName() {
        return name;
    }

    public Set<String> getPrecondition() {
        return precondition;
    }

    public Set<String> getPositiveEffect() {
        return positiveEffect;
    }

    public Set<String> getNegativeEffect() {
        return negativeEffect;
    }
}

File: upf/cpp/action.cxx
#include "action.hxx"

namespace upf
{

  Action::Action(const std::string& name,
                 const std::set<std::string>& precondition,
                 const std::set<std::string>& positive_effect,
                 const std::set<std::string>& negative_effect)
  {
    name_ = name;
    precondition_ = precondition;
    positive_effect_ = positive_effect;
    negative_effect_ = negative_effect;
  }

  const std::string& Action::name() const
  {
    return name_;
  }

  const std::set<std::string>& Action::precondition() const
  {
    return precondition_;
  }

  const std::set<std::string>& Action::positive_effect() const
  {
    return positive_effect_;
  }

  const std::set<std::string>& Action::negative_effect() const
  {
    return negative_effect_;
  }

  bool Action::operator<(const Action& oth) const
  {
    return name_ < oth.name();
  }

}
